### ISSUE TITLE
plotly_r has deprecated `orca` and now uses `kaleido`

### ISSUE THREAD

**Issue by cysouw** [LaurentRDC/pandoc-plot#71]
I was just experimenting for the first time with this phenomenal filter and noted the following problems: 

- the plotly library in R has deprecated the `orca` function for static image export and replaced it with `kaleido`.
-  when using format=HTML the htmlwidget is taking over the whole page, not showing any of the remaining markdown document.

Just for reference: this is the markdown I was experimenting with (macOS 14.5, R 4.4.0):

```
# Test

This is a test

```{.ggplot2 format=SVG}
ggplotify::as.ggplot(~
  plot(1:10,10:1)
)
```

More text

```{.plotly_r format=HTML caption="plotly widget"}
library(plotly)
plot_ly(economics, x = ~pop)
```

blablabla
```

**Comment by LaurentRDC** [comment]
Let me tackle the first point, orca deprecation.

Under the hood, `pandoc-plot` extends your script until it looks like this:

```r
pdf(NULL) # prevent the creation of an empty pdf

# Start of the user plot
library(plotly)
plot_ly(economics, x = ~pop)

# Capture the plot
library(plotly) # just in case
if (!require("processx")) install.packages("processx")
orca(last_plot(), file = "<filename>")
```

I'm not familiar with the R programming language and the `plotly` library; how should the script above be modified?

---

Regarding your second point, exporting to HTML. The script will be extended slightly differently, to this:

```r
# Start of the user plot
library(plotly)
plot_ly(economics, x = ~pop)

# Capture the plot
library(plotly) # just in case
library(htmlwidgets)
p <- last_plot()
htmlwidgets::saveWidget(as_widget(p), "<filename.html>")
```

How should this script be modified so that the widget is not taking over the whole page?

**Comment by LaurentRDC** [comment]
[Plotly's documentation still refers to orca](https://plotly.com/r/static-image-export/); where can I see this deprecation notice?

**Comment by cysouw** [comment]
Indeed, sorry. In the R-help it says "superseded by kaleido", but orca should still work. I'll try to get it running on my side, sorry.

https://www.rdocumentation.org/packages/plotly/versions/4.10.4/topics/orca

**Comment by cysouw** [comment]
As for the Plotly-html output: what you are currently doing will produce a standalone HTML page. I have not been able to find what the Rmarkdown people are doing to include them into the markdown. I will research further, but I'm also not a specialist in such problem :-( 

all the best with this project! I really like this approach.

**Comment by cysouw** [comment]
A possible solution is to embed the resulting html-file into an iframe. Pandoc should be able to just include such a html-tag:

<iframe src = "<filename>"></iframe>

**Comment by cysouw** [comment]
some more research turned up this suggestion: `htmltools::tags$iframe(src = "filename")`

https://plotly-r.com/saving

**Comment by LaurentRDC** [comment]
Awesome, thanks for taking the time to do the research.

I don't have the time to work on this these days, but I am happy to review pull requests
